inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/extensions/DynamicWorkflowDocumentNavigationActionBarExtension.java

```diff
@@ -0,0 +1,120 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package de.tudarmstadt.ukp.inception.workload.dynamic.extensions;
+
+
+import org.apache.wicket.ajax.AbstractAjaxTimerBehavior;
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.markup.html.panel.Panel;
+import org.apache.wicket.util.time.Duration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import de.tudarmstadt.ukp.clarin.webanno.api.DocumentService;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.actionbar.ActionBarExtension;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.actionbar.docnav.DefaultDocumentNavigatorActionBarExtension;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.page.AnnotationPageBase;
+import de.tudarmstadt.ukp.clarin.webanno.model.AnnotationDocument;
+import de.tudarmstadt.ukp.clarin.webanno.model.SourceDocument;
+import de.tudarmstadt.ukp.inception.workload.dynamic.manager.WorkflowProperties;
+import de.tudarmstadt.ukp.inception.workload.dynamic.manager.WorkflowPropertiesImpl;
+import de.tudarmstadt.ukp.inception.workload.dynamic.support.AnnotationQueueOverviewDataProvider;
+
+
+@Order(1000)
+@Component
+public class DynamicWorkflowDocumentNavigationActionBarExtension implements ActionBarExtension
+{
+
+    private final WorkflowProperties workflowProperties;
+    private final DocumentService documentService;
+
+    @Autowired
+    public DynamicWorkflowDocumentNavigationActionBarExtension(
+        WorkflowPropertiesImpl aWorkflowProperties, DocumentService aDocumentService)
+    {
+        workflowProperties = aWorkflowProperties;
+        documentService = aDocumentService;
+    }
+
+    @Override
+    public String getRole()
+    {
+        return DefaultDocumentNavigatorActionBarExtension.class.getName();
+    }
+
+    @Override
+    public int getPriority()
+    {
+        if (workflowProperties.isWorkflowManagerActive())
+        {
+            return 1;
+        } else {
+            return -1;
+        }
+    }
+
+    @Override
+    public Panel createActionBarItem(String aId, AnnotationPageBase aPage)
+    {
+        return new DynamicDocumentNavigator(aId);
+    }
+
+
+
+    //Init of the page, select a random document
+    @Override
+    public void onInitialize(AnnotationPageBase aPage)
+    {
+        AnnotationQueueOverviewDataProvider prov =
+            new AnnotationQueueOverviewDataProvider(
+                documentService.listAnnotationDocuments
+            (aPage.getModelObject().getProject()), documentService.
+                listSourceDocuments(aPage.getModelObject().getProject()), documentService);
+        SourceDocument doc = prov.getRandomDocument(aPage, new AnnotationDocument());
+        if (doc == null)
+        {
+            aPage.info("No more documents available");
```

Comment:
Feedback message won't be displayed if you immediately redirect to another page afterwards.